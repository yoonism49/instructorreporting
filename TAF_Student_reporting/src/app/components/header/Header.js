// @flow weak

import React        from 'react';
import PropTypes    from 'prop-types';
import UserMenu     from './userMenu/UserMenu';
import TaskMenu     from './taskMenu/TaskMenu';
import MessageMenu  from './messageMenu/MessageMenu';
import Button       from './button/Button';
import logo         from '../../img/logo.jpg';

const Header = ({
  userLogin,
  userFirstname,
  userLastname,
  userPicture,
  showPicture,
  appName,
  toggleSideMenu,
  onLogout
}) => (
  <header
    className="header fixed--header">
    <a href="#"
      className="logo">
      <img src={logo}/>
    </a>
    <nav
      className="navbar navbar-static-top"
      role="navigation">
      <div className="navbar-title">Reports Portal</div>
      <div className="navbar-right">
        <ul className="nav navbar-nav">
          <MessageMenu />
          <TaskMenu />
          <UserMenu
            login={userLogin}
            firstname={userFirstname}
            lastname={userLastname}
            picture={userPicture}
            showUserPicture={showPicture}
            onLogout={onLogout}
          />
        </ul>
      </div>
    </nav>
  </header>
);

Header.propTypes = {
  appName:        PropTypes.string,
  userLogin:      PropTypes.string,
  userFirstname:  PropTypes.string,
  userLastname:   PropTypes.string,
  userPicture:    PropTypes.string,
  showPicture:    PropTypes.bool,
  onLogout:       PropTypes.func,

  currentView:    PropTypes.string,
  toggleSideMenu: PropTypes.func
};

Header.defaultProps = {
  appName: 'applicationName'
};

Header.displayName = 'Header';

export default Header;
