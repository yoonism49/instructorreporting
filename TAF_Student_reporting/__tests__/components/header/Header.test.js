
'use strict';

import React          from 'react';
import Header         from '../../../src/app/components/header/Header';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

// -------------------------------------------------
// no more needed image assets hack see package.json better solution:
// -------------------------------------------------
// "jest": {
//   "moduleNameMapper": {
//     "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
//     "\\.(css|less|sass|scss)$": "identity-obj-proxy"
//   }
// }

// jest.mock(
//   '../../../src/app/vendors/img/26115.jpg',
//   () => ''
// );
// jest.mock(
//   '../../../src/app/vendors/img/avatar.png',
//   () => ''
// );

describe('Header component', () => {
  const mockToggleSideMenu = jest.fn();

  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <Header
          toggleSideMenu={mockToggleSideMenu}
        />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
