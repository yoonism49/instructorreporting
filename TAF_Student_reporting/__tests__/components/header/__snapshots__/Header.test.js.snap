// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders as expected 1`] = `
<div>
  <header
    className="header fixed--header"
  >
    <a
      className="logo"
      href="#"
    >
      applicationName
    </a>
    <nav
      className="navbar navbar-static-top"
      role="navigation"
    >
      <a
        className="navbar-btn sidebar-toggle"
        data-toggle="offcanvas"
        href="#"
        onClick={[Function]}
        role="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </a>
      <div
        className="navbar-right"
      >
        <ul
          className="nav navbar-nav"
        >
          <li
            className="dropdown messages-menu"
          >
            <a
              className="dropdown-toggle"
              data-toggle="dropdown"
              href="#"
            >
              <i
                className="fa fa-envelope"
              />
              <span
                className="label label-success"
              >
                4
              </span>
            </a>
            <ul
              className="dropdown-menu"
            >
              <li
                className="header"
              >
                You have 4 messages
              </li>
              <li>
                <ul
                  className="menu"
                >
                  <li>
                    <a
                      href="#"
                    >
                      <div
                        className="pull-left"
                      >
                        <img
                          alt="User Image"
                          className="img-circle"
                          src="test-file-stub"
                        />
                      </div>
                      <h4>
                        Support Team
                      </h4>
                      <p>
                        Why not buy a new awesome theme?
                      </p>
                      <small
                        className="pull-right"
                      >
                        <i
                          className="fa fa-clock-o"
                        />
                        5 mins
                      </small>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <div
                        className="pull-left"
                      >
                        <img
                          alt="user image"
                          className="img-circle"
                          src="test-file-stub"
                        />
                      </div>
                      <h4>
                        Director Design Team
                      </h4>
                      <p>
                        Why not buy a new awesome theme?
                      </p>
                      <small
                        className="pull-right"
                      >
                        <i
                          className="fa fa-clock-o"
                        />
                        2 hours
                      </small>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <div
                        className="pull-left"
                      >
                        <img
                          alt="user image"
                          className="img-circle"
                          src="test-file-stub"
                        />
                      </div>
                      <h4>
                        Developers
                      </h4>
                      <p>
                        Why not buy a new awesome theme?
                      </p>
                      <small
                        className="pull-right"
                      >
                        <i
                          className="fa fa-clock-o"
                        />
                        Today
                      </small>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <div
                        className="pull-left"
                      >
                        <img
                          alt="user image"
                          className="img-circle"
                          src="test-file-stub"
                        />
                      </div>
                      <h4>
                        Sales Department
                      </h4>
                      <p>
                        Why not buy a new awesome theme?
                      </p>
                      <small
                        className="pull-right"
                      >
                        <i
                          className="fa fa-clock-o"
                        />
                        Yesterday
                      </small>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <div
                        className="pull-left"
                      >
                        <img
                          alt="user image"
                          className="img-circle"
                          src="test-file-stub"
                        />
                      </div>
                      <h4>
                        Reviewers
                      </h4>
                      <p>
                        Why not buy a new awesome theme?
                      </p>
                      <small
                        className="pull-right"
                      >
                        <i
                          className="fa fa-clock-o"
                        />
                        2 days
                      </small>
                    </a>
                  </li>
                </ul>
              </li>
              <li
                className="footer"
              >
                <a
                  href="#"
                >
                  See All Messages
                </a>
              </li>
            </ul>
          </li>
          <li
            className="dropdown tasks-menu"
          >
            <a
              className="dropdown-toggle"
              data-toggle="dropdown"
              href="#"
            >
              <i
                className="fa fa-tasks"
              />
              <span
                className="label label-danger"
              >
                9
              </span>
            </a>
            <ul
              className="dropdown-menu"
            >
              <li
                className="header"
              >
                You have 9 tasks
              </li>
              <li>
                <ul
                  className="menu"
                >
                  <li>
                    <a
                      href="#"
                    >
                      <h3>
                        Design some buttons
                        <small
                          className="pull-right"
                        >
                          20%
                        </small>
                      </h3>
                      <div
                        className="progress progress-striped xs"
                      >
                        <div
                          aria-valuemax="100"
                          aria-valuemin="0"
                          aria-valuenow="20"
                          className="progress-bar progress-bar-success"
                          role="progressbar"
                          style={
                            Object {
                              "width": "20%",
                            }
                          }
                        >
                          <span
                            className="sr-only"
                          >
                            20% Complete
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <h3>
                        Create a nice theme
                        <small
                          className="pull-right"
                        >
                          40%
                        </small>
                      </h3>
                      <div
                        className="progress progress-striped xs"
                      >
                        <div
                          aria-valuemax="100"
                          aria-valuemin="0"
                          aria-valuenow="20"
                          className="progress-bar progress-bar-danger"
                          role="progressbar"
                          style={
                            Object {
                              "width": "40%",
                            }
                          }
                        >
                          <span
                            className="sr-only"
                          >
                            40% Complete
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <h3>
                        Some task I need to do
                        <small
                          className="pull-right"
                        >
                          60%
                        </small>
                      </h3>
                      <div
                        className="progress progress-striped xs"
                      >
                        <div
                          aria-valuemax="100"
                          aria-valuemin="0"
                          aria-valuenow="20"
                          className="progress-bar progress-bar-info"
                          role="progressbar"
                          style={
                            Object {
                              "width": "60%",
                            }
                          }
                        >
                          <span
                            className="sr-only"
                          >
                            60% Complete
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                  <li>
                    <a
                      href="#"
                    >
                      <h3>
                        Make beautiful transitions
                        <small
                          className="pull-right"
                        >
                          80%
                        </small>
                      </h3>
                      <div
                        className="progress progress-striped xs"
                      >
                        <div
                          aria-valuemax="100"
                          aria-valuemin="0"
                          aria-valuenow="20"
                          className="progress-bar progress-bar-warning"
                          role="progressbar"
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        >
                          <span
                            className="sr-only"
                          >
                            80% Complete
                          </span>
                        </div>
                      </div>
                    </a>
                  </li>
                </ul>
              </li>
              <li
                className="footer"
              >
                <a
                  href="#"
                >
                  View all tasks
                </a>
              </li>
            </ul>
          </li>
          <li
            className="dropdown user user-menu"
          >
            <a
              className="dropdown-toggle"
              data-toggle="dropdown"
              href="#"
            >
              <i
                className="fa fa-user"
              />
              <span>
                 
                <i
                  className="caret"
                />
              </span>
            </a>
            <ul
              className="dropdown-menu dropdown-custom dropdown-menu-right"
            >
              <li
                className="dropdown-header text-center"
              >
                Account
              </li>
              <li>
                <a
                  href="#"
                >
                  <i
                    className="fa fa-clock-o fa-fw pull-right"
                  />
                  <span
                    className="badge badge-success pull-right"
                  >
                    10
                  </span>
                  Updates
                </a>
                <a
                  href="#"
                >
                  <i
                    className="fa fa-envelope-o fa-fw pull-right"
                  />
                  <span
                    className="badge badge-danger pull-right"
                  >
                    5
                  </span>
                  Messages
                </a>
                <a
                  href="#"
                >
                  <i
                    className="fa fa-magnet fa-fw pull-right"
                  />
                  <span
                    className="badge badge-info pull-right"
                  >
                    3
                  </span>
                  Subscriptions
                </a>
                <a
                  href="#"
                >
                  <i
                    className="fa fa-question fa-fw pull-right"
                  />
                  <span
                    className="badge pull-right"
                  >
                    11
                  </span>
                  FAQ
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href="#"
                >
                  <i
                    className="fa fa-user fa-fw pull-right"
                  />
                  Profile
                </a>
                <a
                  data-toggle="modal"
                  href="#modal-user-settings"
                >
                  <i
                    className="fa fa-cog fa-fw pull-right"
                  />
                  Settings
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-ban fa-fw pull-right"
                  />
                  Logout
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</div>
`;
